### ISSUE TITLE
fix(cli): per-file walker errors print Rust Debug structs; BrokenPipe not swallowed

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#665]
## Current behavior

`bca dump -p ok.rs | head -3` prints `Os { code: 32, kind: BrokenPipe, message: "Broken pipe" } for file ok.rs` — verified. Source: src/concurrent_files.rs:169:

```rust
eprintln!("{err:?} for file {}", path.display());
```

Every per-file walker error is Debug-formatted and BrokenPipe is reported like a real failure.

## Why it's a UX problem

Piping to `head`/`less` is routine; printing a Rust `Os { … }` struct for it looks like a crash. It is also inconsistent with the CLI's own policy: `write_stdout_or_die` deliberately tolerates BrokenPipe (big-code-analysis-cli/src/lib.rs:128-136, "the typical case when the consumer is `head`, `less`, etc."). Debug formatting leaks internal type structure into user-facing diagnostics for all other errors too.

## Proposed 2.0 design

In the walker's per-file error path: swallow `ErrorKind::BrokenPipe` silently (matching `write_stdout_or_die`), and Display-format everything else as `error processing <path>: <err>`.

## Breaking

No — stderr diagnostics only.

Found by the 2026-06 follow-up 2.0 UX audit. Rolls up under #505.
Related: #609.

---

## Resolution Plan

No design decision required — behavior is dictated by the CLI's own existing
policy (`write_stdout_or_die` already tolerates BrokenPipe).

### Steps

1. In the walker per-file error path (`src/concurrent_files.rs:169`):
   - **Swallow `ErrorKind::BrokenPipe` silently**, matching
     `write_stdout_or_die` (`big-code-analysis-cli/src/lib.rs:128-136`) — the
     normal `| head`/`| less` case.
   - **Display-format** every other error as `error processing <path>: <err>`
     (not Debug `{:?}`), so internal type structure stops leaking. Route through
     #609's `warn()`/`error:` convention once it lands (bare lowercase prefix).
2. **Recommended refinement (note, not a user decision):** on the first
   `BrokenPipe`, stop the walk rather than swallowing it per-file for every
   remaining file — the consumer is gone, so continuing to process and write is
   wasted work. Mirror however `write_stdout_or_die` terminates.
3. Tests: `bca dump -p ok.rs | head -3` produces no `Os { … }` /`BrokenPipe`
   line; a genuine per-file error (e.g. unreadable file) prints
   `error processing <path>: <display>` and does not abort the whole walk
   (other files still processed).
4. CHANGELOG `## [Unreleased]` under **Fixed** (stderr diagnostics only;
   non-breaking).
5. `make pre-commit`.

### Related

- #609 (diagnostic prefix unification — share the emitter/convention). Rolls up
  under #505.

## Assessment

| Dimension  | Rating |
|------------|--------|
| Difficulty | Low |
| Complexity | Low |
| Priority   | Medium |

---

## Resolution

Fixed on `fix/issue-665` (commit 2d709207).

`src/concurrent_files.rs` per-file error path now swallows
`ErrorKind::BrokenPipe` silently (matching `write_stdout_or_die`) and
Display-formats every other error as `error processing <path>: <err>`
(no Debug struct leak). Decision extracted into pure
`per_file_error_message` helper; two regression tests added in
`src/concurrent_files_tests.rs`, verified via revert.

Step 2 (stop-the-walk on first BrokenPipe) deferred as a follow-up —
it was the plan's "recommended refinement, not a user decision", and a
coordinated cross-worker early-stop is larger than this diagnostic fix.
Library-stderr smell tracked under #705; #609 prefix unification can
adopt the new message later. Non-breaking (stderr only); no submodule
drift. `make pre-commit` passes.

**Comment by dekobon** [comment]
Fixed on `fix/issue-665` (commit 2d709207).

The per-file walker error path (`src/concurrent_files.rs`) now:

- **Swallows `ErrorKind::BrokenPipe` silently**, matching the CLI's
  `write_stdout_or_die` policy (`big-code-analysis-cli/src/lib.rs:128-136`) —
  the routine `| head`/`| less` case.
- **Display-formats** every other error as `error processing <path>: <err>`
  (no more Debug `Os { code, kind, .. }` struct leak).

The format decision was extracted into a pure `per_file_error_message` helper
so it is unit-testable without stderr capture. Two regression tests assert
BrokenPipe -> no output and a non-BrokenPipe `io::Error` -> the Display string
with no Debug struct shape. Both were verified via revert (they fail against
the old `{err:?}` line).

### Notes / out of scope

- **Step 2 (stop the walk on first BrokenPipe)** was the plan's "recommended
  refinement, not a user decision". Left unimplemented to keep the fix
  minimal — the consumer pool processes the terminal file list concurrently,
  so a coordinated early-stop would need a shared signal across workers; that
  is a larger change than the diagnostic fix this issue scopes. Worth a
  follow-up if the wasted post-pipe-close work proves measurable.
- A library writing to stderr at all is a pre-existing smell (#705); not
  expanded here — the fix matches the `eprintln!` already present.
- #609's diagnostic-prefix unification (`warn()`/`error:` emitter) is not yet
  landed; the new message uses a bare lowercase prefix and can be routed
  through that convention when it arrives.

`make pre-commit` passes; no submodule drift (stderr-only change).
